### PR TITLE
Update UI for the add habit dialog box

### LINKED ISSUE #53: Update add habit flow to match UI mockup
Once the story to add a habit is in (#31), we should go in and improve the UI to match the Figma mockup as much as possible

### PR DESCRIPTION
**Pull request by rudy-patel**
### What is it?
This PR is a follow-up to the original add habit story (#31), and is responsible for cleaning up some of the associated code and updating the UI to match the mockup more closely. This PR closes #53!

### Screenshots
New add habit dialog box:
image

### CODE REVIEW

File: app/src/main/java/com/example/habitsmasher/ui/dashboard/AddHabitDialog.java [hunk 1]
```diff
@@ -70,11 +71,14 @@ public void onClick(View view) {
...
             public void onClick(View view) {
                 Log.d(TAG, "Confirm");
+
                 String habitTitleInput = _habitTitleEditText.getText().toString();
                 String habitReasonInput = _habitReasonEditText.getText().toString();
                 String habitDateInput = _habitDateEditText.getText().toString();
+
                 DateFormat inputDateFormatter = new SimpleDateFormat(DATE_FORMAT);
                 Date habitDate = null;
+
```
Comment (rudy-patel) [inline app/src/main/java/com/example/habitsmasher/ui/dashboard/AddHabitDialog.java:81]: Just adding new lines to make it more readable.

File: app/src/main/java/com/example/habitsmasher/ui/dashboard/HabitListFragment.java [hunk 2]
```diff
@@ -56,15 +56,18 @@ public View onCreateView(@NonNull LayoutInflater inflater,
...
 
         initializeRecyclerView(layoutManager, view);
         return view;
     }
 
-    private void openHabitDialog() {
+    /**
+     * This helper method is responsible for opening the add habit dialog box
+     */
+    private void openAddHabitDialogBox() {
```
Comment (rudy-patel) [inline app/src/main/java/com/example/habitsmasher/ui/dashboard/HabitListFragment.java:70]: Renaming this method for clarity

File: app/src/main/res/layout/fragment_habit_list.xml [hunk 3]
```diff
@@ -24,10 +24,12 @@
...
+        android:layout_marginEnd="20dp"
+        android:layout_marginBottom="70dp"
+        android:backgroundTint="@color/habit_add_button"
         android:clickable="true"
         android:focusable="true"
-        android:layout_marginBottom="70dp"
-        android:layout_marginEnd="20dp" />
+        android:src="@android:drawable/ic_input_add"
+        app:layout_constraintBottom_toBottomOf="parent"
+        app:layout_constraintEnd_toEndOf="parent" />
```
Comment (rudy-patel) [inline app/src/main/res/layout/fragment_habit_list.xml:34]: These changes add the **plus** icon that was missing from the floating action button

File: app/src/main/res/values/strings.xml [hunk 4]
```diff
@@ -6,8 +6,15 @@
...
+    <string name="habit_reason">Habit reason</string>
+    <string name="habit_date">Habit date</string>
     <string name="habit_list">Habit List</string>
+    <string name="dd_mm_yyyy">dd-mm-yyyy</string>
+    <string name="confirm">Confirm</string>
+    <string name="cancel">Cancel</string>
+    <string name="name">Name:</string>
+    <string name="reason">Reason:</string>
+    <string name="date_start">Date start:</string>
+    <string name="add_habit">Add Habit</string>
```
Comment (rudy-patel) [inline app/src/main/res/values/strings.xml:19]: Extracted some common XML strings

**Review by jaskim9824 — APPROVED** [review]

**Review by JacobNguyen71 — APPROVED** [review]

**Pull request merged**